cmd/util.go

```diff
@@ -214,25 +214,35 @@ func multiWordReplace(s string, subname string, sub string) string {
 }
 
 func getKubeVersion() string {
+	var ver string
+	var matched bool
+
 	failmsg := "kubernetes version check failed"
 	// These executables might not be on the user's path.
 	_, err := exec.LookPath("kubectl")
 	if err != nil {
-		exitWithError(fmt.Errorf("%s: %s", failmsg, err))
+		continueWithError(err, failmsg)
 	}
 
 	cmd := exec.Command("kubectl", "version", "--short")
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		exitWithError(fmt.Errorf("%s, %s", failmsg, out))
+		continueWithError(fmt.Errorf("%s", out), "")
 	}
 
-	validVersionPttn := `\d.\d`
 	serverVersionRe := regexp.MustCompile(`Server Version: v(\d+.\d+)`)
```

Comment:
Might be a good idea to split this into a separate function so that we can have a unit test which checks that, given example output from kubectl version --short, we get the version we expect

Reply:
I think that will be great.

Reply:
> I don't think we need exhaustive testing on possible kube version results.
  getkubeversion's result is very deterministic and small enough to test:
  either the result matches our simple regular expression or we return defaultKubeVersion.

You know what ignore my comment above. I will move the validation bit out and we can set up tests like you suggested.